model.py

```diff
@@ -0,0 +1,135 @@
+parser.add_argument("--decay", default=None, type=str, help="Learning rate decay")
+parser.add_argument("--label_smoothing", default=0.0, type=float, help="Label smoothing")
+
+def main(args: argparse.Namespace) -> None:
+    # Fix random seeds and threads
+    tf.keras.utils.set_random_seed(args.seed)
+    tf.config.threading.set_inter_op_parallelism_threads(args.threads)
+    tf.config.threading.set_intra_op_parallelism_threads(args.threads)
+
+    for k, v in sorted(vars(args).items()):
+        print(f"{k}={v}")
+
+    # Create logdir name
+    args.logdir = os.path.join("logs", f"{args.experiment_id}")
+
+    # Load data
+    torlakian_data = TorlakianData(args.window, args.alphabet_size, args.clean)
+
+    print("-" * 30)
+    print(f"Experiment {args.experiment_id}")
+    
+    layers_dims = args.layers.split(",")
+    layers_dims = [int(l) for l in layers_dims]
+
+    layers = [
+        tf.keras.layers.Input(shape=[2 * args.window + 1], dtype=tf.int32),
+        tf.keras.layers.Lambda(lambda x: tf.one_hot(x, len(torlakian_data.train.alphabet))),
+        tf.keras.layers.Flatten()
+    ]
+
+    for layer_dim in layers_dims:
+        layers.append(tf.keras.layers.Dense(layer_dim, activation=tf.nn.relu))
+        layers.append(tf.keras.layers.Dropout(args.dropout))
+
+    layers.append(tf.keras.layers.Dense(1, activation=tf.nn.sigmoid))
+
+    model = tf.keras.Sequential(layers)
+
+    learning_rate = args.lr
+    if args.decay is not None:
+        decay_steps = torlakian_data.train.size/ args.batch_size * args.epochs
+        if args.decay == 'linear':
+            learning_rate = tf.optimizers.schedules.PolynomialDecay(
+                initial_learning_rate=args.lr, 
+                decay_steps=decay_steps, 
+                end_learning_rate=args.lr_final
+            )
+        elif args.decay == 'exponential':
+            decay_rate = args.lr_final / args.lr
+            learning_rate = tf.optimizers.schedules.ExponentialDecay(
+                initial_learning_rate=args.lr,
+                decay_steps=decay_steps,
+                decay_rate=decay_rate
+            )
+    
+    optimizer = tf.keras.optimizers.Adam(learning_rate=learning_rate)
+
+    model.compile(
+        optimizer=optimizer,
+        loss=tf.losses.BinaryCrossentropy(label_smoothing=args.label_smoothing),
+        metrics=[tf.metrics.BinaryAccuracy(name="accuracy")],
+    )
+
+    stop_early = tf.keras.callbacks.EarlyStopping(monitor='val_loss', patience=10)
+    tb_callback = tf.keras.callbacks.TensorBoard(args.logdir, histogram_freq=1, update_freq=100, profile_batch=0)
+
+    logs = model.fit(
+        torlakian_data.train.data["windows"], 
+        torlakian_data.train.data["labels"],
+        batch_size=args.batch_size, 
+        epochs=args.epochs,
+        validation_data=(torlakian_data.val.data["windows"], torlakian_data.val.data["labels"]),
+        callbacks=[tb_callback, stop_early],
+    )
+
+    def do_uppercase(index):
```

Comment:
add comment on thsi (why `...test.alphabet`)

Reply:
It was quite confusing, I agree. I parameterized the function and moved to the file `inference.py`. Hopefully it's more readable now.